pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,21 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// Equal checks equality of two statuses. This is useful for testing with
+// cmp.Equal.
+func (s *Status) Equal(x *Status) bool {
+	if s == nil || x == nil {
+		return s.IsSuccess() && x.IsSuccess()
+	}
+	if s.code != x.code {
+		return false
+	}
+	if s.code == Error {
+		return cmp.Equal(s.err, x.err, cmpopts.EquateErrors())
+	}
+	return len(s.reasons) == 0 && len(x.reasons) == 0
```

Comment:
I missed this change. Why are we not comparing reasons anymore?

I'm fine not comparing reasons when status is error (in this case, the reason is pretty much the error). But we should compare them for any other case.

Reply:
Just to keep consistency for the two cases.
https://github.com/kubernetes/kubernetes/blob/634acdb69438d41e3dfe368f8fcb3e4ffb60dd85/pkg/scheduler/framework/interface_test.go#L165-L176

Reply:
Yeah, we should only do that for error